freqtrade/commands/cli_options.py

```diff
@@ -530,4 +579,12 @@ def __init__(self, *args, **kwargs):
         help='Do not print epoch details header.',
         action='store_true',
     ),
+    "cfg": Arg(
+        '--cfg',
+        nargs='+',
+        metavar="KEY=VALUE",
+        action=SetDictFromArgAction,
```

Comment:
I'm not sure on doing this "here" / in this way.

I fear a bit that having this logic as part of argparse Argument parsing will limit how this can be used down the road.
Also, you mention the "Pipeline" approach already - which we have in place in the Configuration class - by having this type of parsing "extracted" from there, it'll split the logic up into pieces.

---

I also think this produces strange / unexpected results in some instances.
for example in the below case (2 merged configs, with --cfg in between)

```
freqtrade trade -s <whatever> --config config1.json --cfg stake_amount:float=0.1 --config json2.json
```

i'd expect the 2nd config to "overwrite" the stake-amount if it's set - which doesn't happen.
That way, it would be consistent (last setting wins) between --cfg and --config.